saleor/webhook/transport/asynchronous/transport.py

```diff
@@ -282,32 +307,69 @@ def trigger_webhooks_async_for_multiple_objects(
                 )
 
             with allow_writer():
-                # Use transaction to ensure EventPayload and EventDelivery are created together, preventing inconsistent DB state.
+                # Use transaction to ensure EventPayload and EventDelivery are created
+                # together, preventing inconsistent DB state.
                 with transaction.atomic():
                     payload = EventPayload.objects.create_with_payload_file(data)
                     deliveries.extend(
                         create_event_delivery_list_for_webhooks(
-                            webhooks=regular_webhooks,
+                            webhooks=legacy_webhooks,
                             event_payload=payload,
                             event_type=event_type,
                         )
                     )
 
     if subscription_webhooks:
-        deliveries.extend(
-            create_deliveries_for_multiple_subscription_objects(
-                event_type=event_type,
-                subscribable_objects=[
-                    webhook_payload_data.subscribable_object
-                    for webhook_payload_data in webhook_payloads_data
-                ],
-                webhooks=subscription_webhooks,
-                requestor=requestor,
-                allow_replica=allow_replica,
-                pre_save_payloads=pre_save_payloads,
-                request_time=request_time,
+        subscribable_objects = [
+            webhook_payload_data.subscribable_object
+            for webhook_payload_data in webhook_payloads_data
+        ]
+        if is_deferred_payload:
+            _deferred_deliveries_to_create = []
+            for subscribable_object in subscribable_objects:
```

Comment:
Can we move this logic to separate function? Like we already have for `create_deliveries_for_multiple_subscription_objects(`. It would be more readable IMO.